xscen/aggregate.py

```diff
@@ -79,38 +82,232 @@ def climatological_mean(
+    #     ds_rolling = ds_rolling.stack(time=("year", "month", "day"))
+    #     # rebuild time coord
+    #     if isinstance(ds.indexes["time"], pd.core.indexes.datetimes.DatetimeIndex):
+    #         time_coord = pd.to_datetime(
+    #             {
+    #                 "year": ds_rolling.year.values - window + 1,
+    #                 "month": ds_rolling.month.values,
+    #                 "day": ds_rolling.day.values,
+    #             }
+    #         ).to_list()
+    #     elif isinstance(ds.indexes["time"], xr.coding.cftimeindex.CFTimeIndex):
+    #         time_coord = [
+    #             xclim.core.calendar.datetime_classes[ds.time.dt.calendar](
+    #                 y - window + 1, m, d
+    #             )
+    #             for y, m, d in zip(
+    #                 ds_rolling.year.values,
+    #                 ds_rolling.month.values,
+    #                 ds_rolling.day.values,
+    #             )
+    #         ]
+    #     else:
+    #         raise ValueError("The type of 'time' could not be understood.")
+    #     ds_rolling = ds_rolling.drop_vars({"month", "year", "time", "day"})
+    #     ds_rolling = ds_rolling.assign_coords(time=time_coord).transpose("time", ...)
+    #
+    #     concats.extend([ds_rolling])
+    # ds_rolling = xr.concat(concats, dim="time", data_vars="minimal")
+    #
+    # # modify attrs and history
+    # for vv in ds_rolling.data_vars:
+    #     for a in ["description", "long_name"]:
+    #         update_attr(
+    #             ds_rolling[vv], a, _("{window}-year mean of {attr}."), window=window
+    #         )
+    #
+    #     new_history = (
+    #         f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] {window}-year rolling average (non-centered) "
+    #         f"with a minimum of {min_periods} years of data - xarray v{xr.__version__}"
+    #     )
+    #     history = (
+    #         new_history + " \n " + ds_rolling[vv].attrs["history"]
+    #         if "history" in ds_rolling[vv].attrs
+    #         else new_history
+    #     )
+    #     ds_rolling[vv].attrs["history"] = history
+    # if to_level is not None:
+    #     ds_rolling.attrs["cat:processing_level"] = to_level
+    #
+    # return ds_rolling
+    return climatological_op(
+        ds,
+        op="mean",
+        window=window,
+        min_periods=min_periods,
+        stride=interval,
+        periods=periods,
+        rename_variables=False,
+        to_level=to_level,
+        periods_as_dim=False,
+    )
+
+
+@parse_config
+def climatological_op(
+    ds: xr.Dataset,
+    *,
+    op: Union[str, dict] = "mean",
+    window: int = None,
+    min_periods: Union[int, float] = None,
+    stride: int = 1,
+    periods: list = None,
+    rename_variables: bool = True,
+    to_level: str = "climatology",
+    periods_as_dim: bool = False,
+) -> xr.Dataset:
+    """Perform 'op' over the 'year' dimension for given time periods, respecting the temporal resolution of ds.
```

Comment:
I'd be a little bit clearer here that the function creates a `year` dimension. I read this as you need to have an input with `year` already.

Reply:
Changed.